src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1197 @@
+        hidden_states = self.norm2(hidden_states)
+        hidden_states = self.act(hidden_states)
+        return self.dense2(hidden_states)
+
+
+class Beit3PreTrainedModel(PreTrainedModel):
+    """
+    An abstract class to handle weights initialization and a simple interface for downloading and loading pretrained
+    models.
+    """
+
+    config_class = Beit3Config
+    base_model_prefix = "beit3"
+    main_input_name = "input_ids"
+
+    def _init_weights(self, module):
+        """Initialize the weights"""
+        if isinstance(module, (nn.Linear, nn.Conv2d)):
+            # Slightly different from the TF version which uses truncated_normal for initialization
+            # cf https://github.com/pytorch/pytorch/pull/5617
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.Embedding):
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.padding_idx is not None:
+                module.weight.data[module.padding_idx].zero_()
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+        elif isinstance(
+            module,
+            (
+                Beit3ForVisualReasoning,
+                Beit3ForImageTextRetrieval,
+                Beit3ForVisualQuestionAnswering,
+                Beit3ForImageClassification,
+                Beit3ForCaptioning,
+            ),
+        ):
+            module.beit3.text_embedding.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+
+    def _set_gradient_checkpointing(self, module, value=False):
+        if isinstance(module, Beit3Encoder):
+            module.gradient_checkpointing = value
+
+    def reset_parameters(self):
+        self.fc1.reset_parameters()
+        self.fc2.reset_parameters()
+        if self.ffn_layernorm is not None:
+            self.ffn_layernorm.reset_parameters()
+
+
+class Beit3MultiwayFeedForwardNetwork(nn.Module):
+    def __init__(self, module):
+        super().__init__()
+        self.dim = 1
+        self.first = module
+        self.second = copy.deepcopy(module)
+        self.second.reset_parameters()
+
+    def forward(self, hidden_states, split_position=-1):
+        if split_position == -1:
+            return self.first(hidden_states)
+        if split_position == 0:
+            return self.second(hidden_states)
+        text_hidden, image_hidden = torch.split(
+            hidden_states,
+            [split_position, hidden_states.size(self.dim) - split_position],
+            dim=self.dim,
+        )
+        y1, y2 = self.first(text_hidden), self.second(image_hidden)
+        return torch.cat([y1, y2], dim=self.dim)
+
+
+class Beit3Linear(nn.Module):
+    def __init__(self, config):
+        super(Beit3Linear, self).__init__()
```

Comment:
This isn't needed in py3

```suggestion
        super().__init__()
```